### ISSUE TITLE
0.4.2 is a breaking change version, case grpc-rust compile failed! 

### ISSUE THREAD

**Issue by TheWaWaR** [stepancheg/rust-http2#23]
For Struct `httpbis::server::Server`

* httpbis version 0.4.1
    ``` rust
    fn local_addr(&self) -> &SocketAddr
    ```
* httpbis version 0.4.2
    ``` rust
    pub fn local_addr(&self) -> &AnySocketAddr
    ```

This cause `grpc-rust` compile failed.

```
error[E0308]: mismatched types
   --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/grpc-0.2.1/src/server.rs:125:9
    |
125 |         self.server.local_addr()
    |         ^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `std::net::SocketAddr`, found enum `httpbis::socket::AnySocketAddr`
    |
    = note: expected type `&std::net::SocketAddr`
               found type `&httpbis::socket::AnySocketAddr`

error: aborting due to previous error
```

**Comment by stepancheg** [comment]
My bad! Going to publish 0.4.3 to fix compilation.

At the moment you can override local version of rust-http2 to use previous version.

**Comment by stepancheg** [comment]
Published v0.4.3. Should work now. Thanks!

**Comment by weiznich** [comment]
This continues to be a breaking change.
Given the following code:

```rust
let socket_addr = IpAddr::from_str("127.0.0.1:8000").unwrap();

let mut server = ServerBuilder::<TlsAcceptor>::new();
server.http.addr = Some(socket_addr);
```

the type of `server.http.addr` changes from `std::net::SocketAddr` to `httpbis::socket::AnySocketAddr`.
